### PR TITLE
Auto configure Rx OTA protocol from the Tx

### PR DESCRIPTION
**Pull request by JyeSmith**
This is bit of a POC for changing the rx OTA protocol via the Tx.

This then opens up the question of should the Tx lua also contain the Protocol1 list and be used to configure the Rx?

Or should only OTA changes be sent, and the serial output config left on the Rx?  In which case is another config setting required to restore the original non-mavlink protocol when changing back to Normal from Mavlink? Or is defaulting back the crsf ok?

### CODE REVIEW

File: src/src/rx_main.cpp [hunk 1]
```diff
@@ -1044,7 +1044,7 @@ static void ICACHE_RAM_ATTR updateSwitchModePendingFromOta(uint8_t newSwitchMode
 static bool ICACHE_RAM_ATTR ProcessRfPacket_SYNC(uint32_t const now, OTA_Sync_s const * const otaSync)
 {
     // Verify the first two of three bytes of the binding ID, which should always match
```
Comment (wvarty) [inline src/src/rx_main.cpp:1046]: I think the comment here needs to change. No longer 2 bytes.

Comment (JyeSmith) [reply src/src/rx_main.cpp:1046]: Yep, and there are a few other places that need cleaning up re magic numbers etc.  Once https://github.com/ExpressLRS/ExpressLRS/pull/2932 is merged and the overall functionality has been decided Ill clean it all up.

**Comment by wvarty** [comment]
We (devs) change this setting a lot, but I bet most users would set the RX serial config once, and then rarely ever again... I think defaulting back to `CRSF` when switching from `MAVLink` back to `Normal` is probs fine...

File: src/src/rx_main.cpp [hunk 2]
```diff
@@ -1059,6 +1059,24 @@ static bool ICACHE_RAM_ATTR ProcessRfPacket_SYNC(uint32_t const now, OTA_Sync_s
...
 #endif
 
+    if (otaSync->otaProtocol) // Normal = 0, MAVLink = 1
+    {
+        config.SetSerialProtocol(PROTOCOL_MAVLINK);
+    }
+    else if (config.GetSerialProtocol() == PROTOCOL_MAVLINK)
+    {
+        config.SetSerialProtocol(PROTOCOL_CRSF); // default back to CRSF
+
```
Comment (pkendall64) [inline src/src/rx_main.cpp:1069]: Remove blank line

**Review by wvarty — APPROVED** [review]

**Review by pkendall64 — COMMENTED** [review]
I think `otaProtocol` can be 1 bit ,unless you are thinking we may have other protocols in future.
In which case we should have enum defined as well and the code for checking on the RX and sending on the RX.
Also, `otaProtocol` on the TX side is set from `config.linkMode`, which is 2 bits, so I think we need some consistency here.

**Comment by JyeSmith** [comment]
> I think `otaProtocol` can be 1 bit ,unless you are thinking we may have other protocols in future. 

Changed to 2 bits.  I want to future proof a little while the bits are still unused.

**Review by pkendall64 — APPROVED** [review]

**Pull request merged**